app/main/lib/elastic_crud.py

```diff
@@ -0,0 +1,99 @@
+import copy
+import uuid
+import json
+from flask import current_app as app
+from app.main.lib.presto import Presto, PRESTO_MODEL_MAP
+from app.main.lib.elasticsearch import store_document, get_by_doc_id
+
+def _after_log(retry_state):
+    app.logger.debug("Retrying image similarity...")
+
+def get_object_by_doc_id(doc_id):
+    return get_by_doc_id(doc_id)
+
+def get_object(task, _):
+    doc_id = task.get("doc_id", None)
+    language = task.get("language", None)
+    context = task.get("context", {})
+    if context:
+        task["contexts"] = [context]
+    store_document(task, doc_id, language)
+    if task.get("content") and not task.get("text"):
+        task["text"] = task["content"]
+    return task, False
+
+def get_context_for_search(task):
+    context = {}
+    dup = copy.deepcopy(task)
```

Comment:
why do we make a copy here?

Reply:
because if not, the pop mutates the task and affects it outside this function call, which is not what we want to do